app/assets/javascripts/bastion/components/bst-bookmark.directive.js

```diff
@@ -0,0 +1,57 @@
+/**
+* @ngdoc directive
+* @name Bastion.components.directive:bookmark
+* @description
+* Provides the bookmarked items for a dropdown menu
+*/
+angular.module('Bastion.components').directive('bstBookmark', function () {
+    return {
+        scope: {
+            controllerName: '=',
+            query: '='
+        },
+        templateUrl: 'components/views/bst-bookmark.html',
+        controller: ['$scope', '$http', function ($scope, $http) {
+            $scope.newBookmark = {};
+
+            $http({
+                method: 'GET',
+                url: '/api/v2/bookmarks',
+                params: {search: 'controller=' + $scope.controllerName}
```

Comment:
We should be using a [`$resource`](https://docs.angularjs.org/api/ngResource/service/$resource) here rather than these `$http` calls.